actions/restore.py

```diff
@@ -227,7 +237,13 @@ def dismantle_cluster():
     for name, data in etcdctl.member_list(endpoint).items():
         if name != my_name:
             log('Disconnecting {}'.format(name), hookenv.DEBUG)
-            etcdctl.unregister(data['unit_id'], endpoint)
+            for _ in range(5):
+                try:
+                    etcdctl.unregister(data['unit_id'], endpoint)
+                    break
+                except EtcdCtl.CommandFailed:
+                    # Randomized back-off timer to let cluster settle
+                    time.sleep(random.randint(1, 3))
```

Comment:
i like the idea of backoff more than i like random here.  If you want to 
sleep based on range loop variable rather than `randint`, that could work out better

```suggestion
                    time.sleep(_+1)  
```